gef.py

```diff
@@ -11056,7 +11060,12 @@ def sync(self, src: str, dst: Optional[str] = None) -> bool:
             return True
         tgt.parent.mkdir(parents=True, exist_ok=True)
         dbg(f"[remote] downloading '{src}' -> '{tgt}'")
-        gdb.execute(f"remote get '{src}' '{tgt.absolute()}'")
+
+        try:
+            gdb.execute(f"remote get '{src}' '{tgt.absolute()}'")
```

Comment:
In general we prefer moving exception propagating because it contains more info valuable when debugging (i.e. `gef` debug mode on). I think here it could be ok as-is, but would then need to add a `dbg()` message before `return`-ing False

Reply:
> In general we prefer letting exception propagating because it contains more info valuable when debugging (i.e. `gef` debug mode on). I think here it could be ok as-is, but would then need to add a `dbg()` message before `return`-ing False

I think the `sync` function was initially meant to return false if the sync failed (according to previous code + function signature) but gdb is raising an exception if it fails. I thought it was better to keep the previous design since the code was designed to use the return value instead of an exception.

That's actually why the bug using rr was so terrible, there was an exception here that was not catched at all, because the code was not designed this way at all.

If you want to change the design, that would be totally understandable of course

Reply:
I think returning false here is fine. If we want to switch it to passing the exception to the called (`GefRemoteCommand`) that's fine, but that changes the interface of session managers (I'll have to change the one for black magic probes in gef-extras).